rust_team_data/src/email_encryption.rs

```diff
@@ -0,0 +1,111 @@
+//! This module implements the encryption scheme used to safely include private email addresses in
+//! the team repository. It generates encrypted content that looks like this:
+//!
+//! encrypted+a371d1d65dcc697d000826e4bd59dcbfb16ad73eceb1139caaaab795737bb3a54403f0ec69f3d40c8965a5f4bc52f67c94ec538a8a2fea710cabdaaf27661f9a62b0121db7d229@rust-lang.invalid
+//!
+//! The hex-encoded part of the email address is a concatenation of a 12-byte random nonce and the
+//! AES256-GCM-encrypted email address. Utilities are provided to both encrypt and decrypt.
+
+use aes_gcm::aead::{generic_array::GenericArray, Aead, NewAead};
+use aes_gcm::Aes256Gcm;
+
+const PREFIX: &str = "encrypted+";
+const SUFFIX: &str = "@rust-lang.invalid";
+const KEY_LENGTH: usize = 32;
+const NONCE_LENGTH: usize = 12;
+
+/// Encrypt an email address with the provided key.
+pub fn encrypt(key: &str, email: &str) -> Result<String, Error> {
+    // Generate a random nonce every time something is encrypted.
+    let mut nonce = [0u8; NONCE_LENGTH];
+    getrandom::getrandom(&mut nonce).map_err(Error::GetRandom)?;
+    let nonce = GenericArray::from_slice(&nonce);
+
+    let mut encrypted = init_aes(key)?
+        .encrypt(&nonce, email.as_bytes())
+        .map_err(|_| Error::EncryptionFailed)?;
+
+    // Concatenate both the nonce and the payload, as both will be needed for decryption.
+    let mut payload = nonce.to_vec();
```

Comment:
If we're converting to a vec, why the GenericArray::from_slice? Can we just initially create the nonce via `vec!`?

Reply:
`GenericArray` is required by the `aes_gcm` crate's API.